ts/features/ingress/screens/IngressScreen.tsx

```diff
@@ -0,0 +1,136 @@
+import I18n from "../../../i18n";
+import { isMixpanelEnabled as isMixpanelEnabledSelector } from "../../../store/reducers/persistedPreferences";
+import { trackIngressScreen } from "../../../screens/profile/analytics";
+import LoadingScreenContent from "../../../components/screens/LoadingScreenContent";
+import { OperationResultScreenContent } from "../../../components/screens/OperationResultScreenContent";
+import { useIODispatch, useIOSelector } from "../../../store/hooks";
+import { isBackendStatusLoadedSelector } from "../../../store/reducers/backendStatus";
+import { setIsBlockingScreen } from "../store/actions";
+import ModalSectionStatusComponent from "../../../components/SectionStatus/modal";
+import { isMixpanelInitializedSelector } from "../../mixpanel/store/selectors";
+import {
+  trackIngressCdnSystemError,
+  trackIngressServicesSlowDown,
+  trackIngressTimeout
+} from "../analytics";
+
+const TIMEOUT_CHANGE_LABEL = (5 * 1000) as Millisecond;
+const TIMEOUT_BLOCKING_SCREEN = (10 * 1000) as Millisecond;
+
+export const IngressScreen = () => {
+  const isMixpanelInitialized = useIOSelector(isMixpanelInitializedSelector);
+  const isMixpanelEnabled = useIOSelector(isMixpanelEnabledSelector);
+  const dispatch = useIODispatch();
+  const [netInfo, setNetInfo] = useState<NetInfoState>();
+  const [showBlockingScreen, setShowBlockingScreen] = useState(false);
+  const [contentTitle, setContentTitle] = useState(I18n.t("startup.title"));
+
+  useEffect(() => {
+    // `isMixpanelEnabled` mustn't be `false`
+    if (isMixpanelInitialized && isMixpanelEnabled !== false) {
+      trackIngressScreen();
+    }
+  }, [isMixpanelInitialized, isMixpanelEnabled]);
+
+  useEffect(() => {
+    const timeouts: Array<number> = [];
+    timeouts.push(
+      setTimeout(() => {
+        setContentTitle(I18n.t("startup.title2"));
+        timeouts.shift();
+      }, TIMEOUT_CHANGE_LABEL)
+    );
+    timeouts.push(
+      setTimeout(() => {
+        setShowBlockingScreen(true);
+        dispatch(setIsBlockingScreen());
+        timeouts.shift();
+      }, TIMEOUT_BLOCKING_SCREEN)
+    );
+    void fetchNetInfo()
+      .then(info => {
+        if (!info.isConnected) {
+          timeouts.forEach(clearTimeout);
+        }
+        setNetInfo(info);
+      })
+      .catch();
+
+    return () => {
+      timeouts?.forEach(clearTimeout);
+    };
+  }, [dispatch]);
+
+  if (netInfo && !netInfo.isConnected) {
+    return (
+      <OperationResultScreenContent
+        testID="device-connection-lost-id"
+        pictogram="lostConnection"
+        title={I18n.t("startup.connection_lost.title")}
+        subtitle={I18n.t("startup.connection_lost.description")}
+      />
+    );
+  }
+
+  if (showBlockingScreen) {
+    return <IngressScreenBlockingError />;
+  }
+
+  return (
+    <>
+      <ModalSectionStatusComponent
+        sectionKey="ingress"
+        sticky
+        trackingAction={trackIngressServicesSlowDown}
```

Comment:
This tracking too must be synchronized with mixpanel initialization. Otherwise we'd happen that we try sending it before having a mixpanel instance set up. What do you think?